### PR TITLE
feat: add Gno Playground

### PR DESCRIPTION
**Pull request by ilgooz**

### CODE REVIEW

File: src/tutorials/gno.land/gbe/07-loops/index.ts [hunk 1]
```diff
@@ -2,36 +2,7 @@ const markdownContent: string = `
...
-
-		// you have to be sure to reach the break condition,
-		// if not you will exaust all the available gas for this execution
-		loops++
-	}
-
-	return out
-}
-\`\`\`
+<Playground open="src/tutorials/gno.land/gbe/07-loops/loops.gno" files="%5B%7B%22path%22%3A%22src%2Ftutorials%2Fgno.land%2Fgbe%2F07-loops%2Floops.gno%22%2C%22content%22%3A%22const%20totalLoops%20%3D%2010%5Cn%5Cnfunc%20forLoop()%20string%20%7B%5Cn%5Ctvar%20out%20string%5Cn%5Ctfor%20i%20%3A%3D%200%3B%20i%20%3C%20totalLoops%3B%20i%2B%2B%20%7B%5Cn%5Ct%5Ctout%20%3D%20out%20%2B%20fmt.Sprintf(%5C%22%25d%5C%22%2C%20i)%5Cn%5Ct%7D%5Cn%5Cn%5Ctreturn%20out%5Cn%7D%5Cn%5Cnfunc%20whileLoop()%20string%20%7B%5Cn%5Ctvar%20loops%20int%5Cn%5Ctvar%20out%20string%5Cn%5Ctfor%20%7B%5Cn%5Ct%5Ctif%20loops%20%3D%3D%20totalLoops%20%7B%5Cn%5Ct%5Ct%5Ctbreak%5Cn%5Ct%5Ct%7D%5Cn%5Cn%5Ct%5Ctout%20%3D%20out%20%2B%20%5C%22A%5C%22%5Cn%5Cn%5Ct%5Ct%2F%2F%20you%20have%20to%20be%20sure%20to%20reach%20the%20break%20condition%2C%5Cn%5Ct%5Ct%2F%2F%20if%20not%20you%20will%20exaust%20all%20the%20available%20gas%20for%20this%20execution%5Cn%5Ct%5Ctloops%2B%2B%5Cn%5Ct%7D%5Cn%5Cn%5Ctreturn%20out%5Cn%7D%22%7D%5D">
```
Comment (moul) [inline src/tutorials/gno.land/gbe/07-loops/index.ts:5]: we should keep a way to declare source codes more like it was the case previously.

what about creating a real `loops.gno` file in git, and then find a way to inject it?  this way we'll have real .gno files that we can run with the `gno binary` too.

Comment (ilgooz) [reply src/tutorials/gno.land/gbe/07-loops/index.ts:5]: I think what you are describing is already case, please [see](https://github.com/gnolang/gno-by-example/tree/feat/playground/src/tutorials/gno.land/gbe/07-loops), gno files are there.

`index.ts` contains the generated code from the gno files by using the meta information in the readme file. I believe the generated file doesn't need to be human readable, right?

Comment (zivkovicmilos) [reply src/tutorials/gno.land/gbe/07-loops/index.ts:5]: @ilgooz is right

This `index.ts` is something generated by a script, basically glue code we import when the user wants to see a specific tutorial, so it doesn't necessarily need to be human readable.

The tutorials (gno files) themselves are untouched, and in fact we run full on tests for them in the CI to make sure they are valid and passing

Just for comparison, Solidity by Example encodes these sorts of glue code imports in base64

Comment (moul) [reply src/tutorials/gno.land/gbe/07-loops/index.ts:5]: Ok, sounds good. Thank you for the clarification.

Please, ping me for the next PR adding an exercise, so I can check if the review looks reasonable.

Isn't it a good practice to name generated files with `.gen.ts`? We could set this extension as `linguist-generated` in `.gitattributes` to further ease the review process.

Comment (moul) [reply src/tutorials/gno.land/gbe/07-loops/index.ts:5]: Ok, sounds good. Thank you for the clarification.

Please, ping me for the next PR adding an exercise, so I can check if the review looks reasonable.

Isn't it a good practice to name generated files with `.gen.ts`? We could set this extension as `linguist-generated` in `.gitattributes` to further ease the review process.

File: src/components/atoms/Playground/index.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    editorBase: 'vs-dark',
+  } 
+
+  if (isLight) {
+    theme = {
+      color: "#606060",
+      primaryColor: '#f1f1f1',
+      secondaryColor: '#bebebe',
+      editorBase: 'vs',
+    }
```
Comment (zivkovicmilos) [inline src/components/atoms/Playground/index.tsx:27]: You can probably use the already predefined Chakra theme color constants from `src/theme/theme.ts`, as I can see some of them match

File: scripts/generate.ts [hunk 3]
```diff
@@ -101,22 +102,26 @@ const parseTutorial = (directory: string): TutorialItem => {
...
+
+    filePathDependsOn.forEach(function(filePath) {
+      const codeFilePath = path.join(directory, filePath);
+      const codeContent = fs.readFileSync(codeFilePath, 'utf-8');
+      files.push({ path: codeFilePath, content: codeContent })
+    })
+
+    const filesEncoded = encodeURIComponent(JSON.stringify(files)) 
+
+    const embeddedCodeBlock = `<Playground open="${codeFilePath}" files="${filesEncoded}">`;
```
Comment (zivkovicmilos) [inline scripts/generate.ts:120]: Is the prop `open` leftover? I don't see it used in the`Playground` component

**Review by zivkovicmilos — APPROVED** [review]
Looks amazing 👏 

Thank you and the rest of the team for pushing this out on such short notice

I've left a few minor comments, otherwise it looks good 💯

File: scripts/generate.ts (hunk 3 above)
Comment (zivkovicmilos) [reply scripts/generate.ts:120]: I would also add a sentence or two about the `depends_on_file` syntax that was added in the repo README

Comment (ilgooz) [reply scripts/generate.ts:120]: * `open` will be used in a future version playground, let me pass it to the component. 
* docs added by 61c1541

File: src/components/atoms/Playground/index.tsx (hunk 2 above)
Comment (ilgooz) [reply src/components/atoms/Playground/index.tsx]: done by 61c1541

**Pull request merged**